Mage/Task/AbstractTask.php

```diff
@@ -307,4 +308,56 @@ protected function untarRelease($releaseId)
         }
         return $result;
     }
+
+    /**
+     * Returns the array of environment variables
+     * Returned array contains both system variables and variables set in config
+     * WARNING: To access system's variables you need to set proper value in your php.ini at variables_order key
+     * @see http://php.net/manual/en/ini.core.php#ini.variables-order
+     *
+     * @return array
+     */
+    protected function getEnvVariables()
+    {
+        $configVars = array_merge(
+            $this->getConfig()->general('env', []),
+            $this->getConfig()->environmentConfig('env', []),
+            $this->getConfig()->getParameter('env', []),
```

Comment:
This line should be : `$this->getParameter('env', array())` in order to retrieve actual task config parameter. Same goes for line 327 for the `env.variables` shortcut.